shared/composemanager/compose_manager.go

```diff
@@ -0,0 +1,170 @@
+package composemanager
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/chanzuckerberg/happy/shared/config"
+	"github.com/compose-spec/compose-go/loader"
+	"github.com/compose-spec/compose-go/types"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	platform_architecture = "platform_architecture"
+	build                 = "build"
+	services              = "services"
+)
+
+type ComposeManager struct {
+	HappyConfig *config.HappyConfig
+}
+
+func NewComposeManager() ComposeManager {
+	return ComposeManager{}
+}
+
+func (c ComposeManager) WithHappyConfig(happyConfig *config.HappyConfig) ComposeManager {
+	c.HappyConfig = happyConfig
+	return c
+}
+
+func (c ComposeManager) Generate(ctx context.Context) error {
+	p := types.Project{}
+	p.Services = types.Services{}
+
+	stackDef, err := c.HappyConfig.GetStackConfig()
+	if err != nil {
+		return errors.Wrap(err, "unable to get stack config")
+	}
+
+	_, ok := stackDef[services]
+	if !ok {
+		return errors.New("unable to find services in stack config")
+	}
+
+	servicesDef := stackDef[services].(map[string]any)
+	if len(servicesDef) == 0 {
+		return errors.New("no service settings are defined in stack config")
+	}
+
+	for _, service := range c.HappyConfig.GetData().Services {
+		if sd, ok := servicesDef[service]; ok {
```

Comment:
```suggestion
		if sd, ok := servicesDef[service]; !ok {
		  continue
		}
		
```

Reply:
https://github.com/uber-go/guide/blob/master/style.md#reduce-nesting

Reply:
This is not workable, as `sd` comes out of scope, and alternative is not super elegant.

Reply:
remove the nesting

Reply:
Done